QgisModelBaker/libqgsprojectgen/generator/generator.py

```diff
@@ -89,13 +89,25 @@ def layers(self, filter_layer_list=[]):
         layer_uri = db_factory.get_layer_uri(self.uri)
         layer_uri.pg_estimated_metadata = self.pg_estimated_metadata
 
+        # When in PostGIS mode, there can be multiple geometries per table - this leads to multiple layers
+        table_appearances = {}
+        check_tables_info=self.get_tables_info()
```

Comment:
Wouldn't this potentially cost a lot?
We have a `tables_info` already in line 84.

Reply:
We cannot iterate through it twice. But agreeing that it would make sense to get it only once and put it into a copy when we use it more than one time after.

Reply:
https://github.com/opengisch/QgisModelBaker/blob/master/QgisModelBaker/libqgsprojectgen/dbconnector/pg_connector.py#L263
Here we have the `yield` returning a generator what means we have to call `self.get_tables_info()` each time we want that information. I think `mssql` and `gpkg` do return an ordinary list. I think it's memory against CPU. @m-kuhn since you introduced this function: Why you use yield here? (beside that it's an awesome functionality :-)) Should it be preferred even when we could reuse the ordinary list? 

Possibilities:
- using `yield` generator and call the `self.get_tables_info()` multiple times.
- using `yield` generator and call it once, but store the result in a list (in the project generator) and reuse it there.
- returning an ordinary list

Reply:
> Why you use yield here?

Probably because it's an awesome functionality that I wanted to try. I think. And apparently it worked stable enogh over the years.

Caching this (i.e. storing it in a list) sounds good.
I'd probably go for option 2 because this keeps the efficient implementation using `yield` available and because I cannot see any advantage of changing this to 3. But this is no strong opinion.